physical/raft/fsm.go

```diff
@@ -352,7 +352,9 @@ func (f *FSM) List(ctx context.Context, prefix string) ([]string, error) {
 				keys = append(keys, key)
 			} else {
 				// Add truncated 'folder' paths
-				keys = strutil.AppendIfMissing(keys, string(key[:i+1]))
+				if len(keys) == 0 || keys[len(keys)-1] != key[:i+1] {
```

Comment:
Is this safe, given that it's only ever looking at the last element?

Reply:
A question I've asked myself many times, and in fact this PR is my way of asking that.  As far as I can tell, the answer is yes.  The `Seek` we do above is premised on the bucket being ordered.

Reply:
I think the part that keeps tripping me up when I think about this is the first `if` appending `key` to `keys` and the `else` appending `key[:i+1]` to `keys`. I don't know enough about this part of the code to know what those values typically look like, but the fact that they're slightly different, and this check isn't exhaustive, made me pause.